argocd/resource_argocd_cluster.go

```diff
@@ -263,8 +230,9 @@ func resourceArgoCDClusterUpdate(ctx context.Context, d *schema.ResourceData, me
 	}
 
 	tokenMutexClusters.Lock()
-	_, err = client.Update(ctx, &clusterClient.ClusterUpdateRequest{Cluster: cluster})
-	tokenMutexClusters.Unlock()
+	defer tokenMutexClusters.Unlock()
```

Comment:
Same thing here, the provider will delay all other ops on unrelated clusters even further, as the `readArgoCDCluster` processing time is now included. I believe this will prove to be a problem at scale, WDYT?